### PR TITLE
Matching engine tutorial - add networking troubleshooting and updates to notebook

### PR DESCRIPTION
**Pull request by DanElias**
**REQUIRED:** Add a summary of your PR here, typically including why the change is needed and what was changed. Include any design alternatives for discussion purposes.

<br>
GCP Customers sometimes have an incorrect configuration which gives them errors such as Error 403 status:14 "failed to connect to all addresses" when their VPC and Matching Engine resources configuration is incorrect. This new section will allow them to troubleshoot their issues without the help from a support agent.
<br><br><br>

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [ ] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [ ] Follow the style and grammar rules outlined in the above notebook template.
- [ ] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [ ] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [ ] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

<br>

2. If you are opening a PR for `Community Notebooks` under the [notebooks/community](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/community) folder:
- [x] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/community/CODEOWNERS) file under the `Community Notebooks` section, pointing to the author or the author's team.
- [x] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

<br>

3. If you are opening a PR for `Community Content` under the [community-content](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/community-content) folder:
- [ ] Make sure your main `Content Directory Name` is descriptive, informative, and includes some of the key products and attributes of your content, so that it is differentiable from other content
- [ ] The main content directory has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/community-content/CODEOWNERS) file under the `Community Content` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

### CODE REVIEW

File: notebooks/community/matching_engine/matching_engine_for_indexing.ipynb [hunk 1]
```diff
@@ -261,7 +261,7 @@
         "\n",
```
Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <p>Should be 2023 if first submission</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <p>old style links. Replace with template version</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <p>Break text cell into 4 text cells.</p><p>Move Dataset to after Objective.</p><p> </p><p>GCP ANN -&gt; Vertex AI Matching Engine</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <pre class="ql-syntax" spellcheck="false">Line #1.    PROJECT_ID = ""  # @param {type:"string"}

use template version for setting PROJECT_ID
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <p>Why preview version?</p><p> </p><p>Vertex SDK -&gt; Vertex AI SDK</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <p>Hum, looks like you set PROJECT_ID twice.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <p>Use template version for REGION and BUCKET_URI</p><p> </p><p>Replace TIMESTAMP with UUID (see template)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <p>Dta -&gt; data (sentence case)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <p>GCS -&gt; Cloud Storage (do not use abbreviations for Cloud products)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <p>Create Indexes -&gt; Create the indexes</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <p>Use style</p><p>Learn more about [title](link)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <p>Update the indexes</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='53'/>

Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <p>Deploy the indexes</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='69'/>

Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <p>Deploy brute force index</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='76'/>

Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <p>Create online queries</p><p> </p><p>GCP -&gt; Google Cloud</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='82'/>

Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <p>Batch Query -&gt; Submit a batch query</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='91'/>

Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <p>Compute the recall metric</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='94'/>

Comment (andrewferlitsch) [inline notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:261]: <p>what about removing the bucket you created?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1465/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='97'/>

Comment (DanElias) [reply notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:269]: Fixed

Comment (DanElias) [reply notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:269]: Fixes

Comment (DanElias) [reply notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:288]: <p>Break text cell into 4 text cells - Done</p><p>Move Dataset to after Objective - Done</p><p>GCP ANN -&gt; Vertex AI Matching Engine - Done</p>

Comment (DanElias) [reply notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:288]: <p>Done</p>

Comment (DanElias) [reply notebooks/community/matching_engine/matching_engine_for_indexing.ipynb:344]: <p>Notebook was a bit outdated. Fixed to use the latest GA version of Vertex AI SDK</p>

Comment (DanElias) [reply notebooks/community/matching_engine/matching_engine_for_indexing.ipynb]: <p>Fixed.</p>

Comment (DanElias) [reply notebooks/community/matching_engine/matching_engine_for_indexing.ipynb]: <p>Fixed</p>

Comment (DanElias) [reply notebooks/community/matching_engine/matching_engine_for_indexing.ipynb]: <p>Fixed</p>

Comment (DanElias) [reply notebooks/community/matching_engine/matching_engine_for_indexing.ipynb]: <p>Fixed</p>

Comment (DanElias) [reply notebooks/community/matching_engine/matching_engine_for_indexing.ipynb]: <p>Fixed</p>

Comment (DanElias) [reply notebooks/community/matching_engine/matching_engine_for_indexing.ipynb]: <p>Fixed</p>

Comment (DanElias) [reply notebooks/community/matching_engine/matching_engine_for_indexing.ipynb]: <p>Fixed</p>

Comment (DanElias) [reply notebooks/community/matching_engine/matching_engine_for_indexing.ipynb]: <p>Fixed</p>

Comment (DanElias) [reply notebooks/community/matching_engine/matching_engine_for_indexing.ipynb]: <p>Fixed</p>

Comment (DanElias) [reply notebooks/community/matching_engine/matching_engine_for_indexing.ipynb]: <p>Added</p>

**Comment by DanElias** [comment]
This notebook is referenced in: https://cloud.google.com/vertex-ai/docs/matching-engine/ann-service-overview#key_user_journeys

**Comment by DanElias** [comment]
All comments have been resolved. Awaiting new review @andrewferlitsch

**Review by andrewferlitsch — APPROVED** [review]

**Pull request merged**